### PR TITLE
Adding Servlet to Change User Roles

### PR DESCRIPTION
**Pull request by dynamitechetan**
This is a servlet that is used to change the role of any user
This requires a to be logged in.
can be tested locally on http://127.0.0.1:4000/aaa/changeRoles

Fixes issue #206

### CODE REVIEW

File: src/ai/susi/server/api/aaa/ChangeUserRoles.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public String getAPIPath() {
+        return "/aaa/changeRoles";
+    }
+
+
+    @Override
+    public JSONObject serviceImpl(Query query, HttpServletResponse response, Authorization authorization, final JsonObjectWithDefault permissions) throws APIException {
+
+        JSONObject result = new JSONObject();
+        if (authorization.getBaseUserRole().toString().equals("ADMIN")) {
```
Comment (Fluepke) [inline src/ai/susi/server/api/aaa/ChangeUserRoles.java]: @Orbiter Is this check necessary? As I understood you, it's enough to override `getMinimalBaseUserRole()`.

File: src/ai/susi/server/api/aaa/ChangeUserRoles.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                    serviceList.put(service.getCanonicalName());
+                }
+                result.put("serviceList", serviceList);
+                return result;
+            } else if (query.get("getUserRolePermission", false)) {
+                result.put("userRolePermissions", authorization.getUserRole().getPermissionOverrides());
+                return result;
+            } else {
+                // Accept POST params
+                String userTobeUpgraded = query.get("user", null);
```
Comment (Fluepke) [inline src/ai/susi/server/api/aaa/ChangeUserRoles.java]: Is it possible that an admin (the only admin) downgrades himself to a normal user resulting in a situation without any admin?

File: src/ai/susi/server/api/aaa/ChangeUserRoles.java (hunk 1 above)
Comment (dynamitechetan) [inline src/ai/susi/server/api/aaa/ChangeUserRoles.java]: yes. I tried to set MinimalBaseUserRole as ADMIN, but it didn't worked. So I just wrote another condition to check it.

File: src/ai/susi/server/api/aaa/ChangeUserRoles.java (hunk 2 above)
Comment (dynamitechetan) [inline src/ai/susi/server/api/aaa/ChangeUserRoles.java]: Yes. Its possible now but why will he do so. 
Also the first Admin has to make himself the admin by changing the value of userRole in the db/authorization.json.
For example, first orbiter can be the admin by changing the value of his userRole in authorization.json and then he can give rights to other admins.

File: src/ai/susi/server/api/aaa/ChangeUserRoles.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                String upgradedRole = query.get("role", "user");
+                // Validation
+                if(userTobeUpgraded==null){
+                    throw new APIException(400, "Bad username");
+                }
+                if(!DAO.authorization.has(userTobeUpgraded)){
+                    throw new APIException(400, "Username not found");
+                }
+                if(!DAO.userRoles.has(upgradedRole))
```
Comment (dynamitechetan) [inline src/ai/susi/server/api/aaa/ChangeUserRoles.java]: Okay.

**Comment by Orbiter** [comment]
I moved the access right check to AbstractAPIHandler in https://github.com/fossasia/susi_server/commit/ed85d9690f901590a6189b02292293a2b453f1a3

There should never be the need to check the user rights inside the servlet, thats the purpose of the generic authorization.

If you write 'it is not working', then please:
- write down what you expected
- write down how you tested
- write down what was the result

So we can discuss about that.

**Pull request merged**